### DISCUSSION TITLE
Couldn't able to install or approve API access request for PnP solution v4.

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#2386] [Q&A] [answered]
Hi, I have installed PnP modern V4 solution at QA and all API access approvals went well (can see 9 api access approvals) but when i tried to upload the same sollution into the other tenant I can only see the below api access but not all.

please guide us what went wrong.

Snapshot:
![image](https://user-images.githubusercontent.com/111119555/190408955-30ea090f-6828-432d-987a-75ba16cff824.png)

**Comment by FranckyC** [comment]
Hi @GangadharMFord, what version are you using? When you deploy the app from the app catalog, do you see the list of required permissions in the popup window?

![image](https://user-images.githubusercontent.com/5659122/190412831-5052c1e2-f13e-4891-8059-b2289eca16b5.png)

**Comment by GangadharMFord** [comment]
Hi Francky,

I have deployed the solution yesterday and here is the screen shot. I'm using latest version V4.
And I'm trying to deploy the solution at site collection level app catalog. But When uploaded into site collection app catalog, Only see few API accesses in Manage API access page but not all. 

![image](https://user-images.githubusercontent.com/111119555/190414031-a05aee9a-9cde-4a4d-bdfa-5c13447801ba.png)

**Comment by FranckyC** [comment, marked as answer]
The permissions listed here seems to be correct so this seems to be an issue on the SharePoint side. Can you try with a 3rd tenant by any chance? As a workaround, you can also add permissions manually https://pnp.github.io/cli-microsoft365/cmd/spo/serviceprincipal/serviceprincipal-grant-add/

**Comment by GangadharMFord** [comment]
thank you Francky C for given information, I checked with global admin where they can see the remaining permissions and able to approve it.

Seems the approver need little more higher permissions to approve some Web api access permissions at API access page.

@Another question, these Web api access permissions will work with delegated mode or App permissions at backend ?

thank you again for help me in this.

**Comment by wobba** [comment]
They are delegated only.
